File: frontend/prisma/seed.ts
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

async function main() {
  const count = await prisma.validationRule.count();
  if (count > 0) {
    console.log("Rules already seeded");
    return;
  }

  await prisma.validationRule.createMany({
    data: [
      { countryName: "India", countryCode: "+91", fieldName: "phone", validationType: "phone_length", ruleValue: "10" },
      { countryName: "Singapore", countryCode: "+65", fieldName: "phone", validationType: "phone_length", ruleValue: "8" },
      { countryName: "Global", countryCode: "", fieldName: "email", validationType: "email_format", ruleValue: "standard" },
      { countryName: "Global", countryCode: "", fieldName: "payment_method", validationType: "enum", ruleValue: "UPI,Card,Wallet,Net Banking,Cash" },
    ],
  });
  console.log("Seeded default validation rules");
}

main()
  .catch(console.error)
  .finally(() => prisma.$disconnect());

File: backend/seed.py
import os
import sys
sys.path.append(os.path.dirname(os.path.abspath(__file__)))

from app.database import SessionLocal, engine
from app.models import Base, ValidationRule

def main():
    Base.metadata.create_all(bind=engine)
    db = SessionLocal()
    try:
        if db.query(ValidationRule).count() == 0:
            db.add_all([
                ValidationRule(country_name="India", country_code="+91", field_name="phone", validation_type="phone_length", rule_value="10"),
                ValidationRule(country_name="Singapore", country_code="+65", field_name="phone", validation_type="phone_length", rule_value="8"),
                ValidationRule(country_name="Global", country_code="", field_name="email", validation_type="email_format", rule_value="standard"),
                ValidationRule(country_name="Global", country_code="", field_name="payment_method", validation_type="enum", rule_value="UPI,Card,Wallet,Net Banking,Cash"),
            ])
            db.commit()
            print("Seeded default validation rules")
        else:
            print("Rules already seeded")
    finally:
        db.close()

if __name__ == "__main__":
    main()
